presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/BenchmarkSTContains.java

```diff
@@ -0,0 +1,128 @@
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkSTContains
+{
+    @Benchmark
+    public Object stContainsInnerPoint(BenchmarkData data)
+    {
+        return GeoFunctions.stContains(data.geometry, data.innerPoint);
+    }
+
+    @Benchmark
+    public Object stContainsInnerPointDeserialized(BenchmarkData data)
+    {
+        return data.ogcGeometry.contains(data.innerOgcPoint);
+    }
+
+    @Benchmark
+    public Object stContainsOuterPointInEnvelope(BenchmarkData data)
+    {
+        return GeoFunctions.stContains(data.geometry, data.outerPointInEnvelope);
+    }
+
+    @Benchmark
+    public Object stContainsOuterPointInEnvelopeDeserialized(BenchmarkData data)
+    {
+        return data.ogcGeometry.contains(data.outerOgcPointInEnvelope);
+    }
+
+    @Benchmark
+    public Object stContainsOuterPointNotInEnvelope(BenchmarkData data)
+    {
+        return GeoFunctions.stContains(data.geometry, data.outerPointNotInEnvelope);
+    }
+
+    @Benchmark
+    public Object stContainsOuterPointNotInEnvelopeDeserialized(BenchmarkData data)
+    {
+        return data.ogcGeometry.contains(data.outerOgcPointNotInEnvelope);
+    }
+
+    @Benchmark
+    public Object deserialize(BenchmarkData data)
+    {
+        return GeometryUtils.deserialize(data.geometry);
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private Slice geometry;
+        private Slice innerPoint;
+        private Slice outerPointInEnvelope;
+        private Slice outerPointNotInEnvelope;
+        private OGCGeometry ogcGeometry;
+        private OGCPoint innerOgcPoint;
+        private OGCPoint outerOgcPointInEnvelope;
+        private OGCPoint outerOgcPointNotInEnvelope;
+
+        @Setup
+        public void setup()
+                throws IOException
+        {
+            Path filePath = Paths.get(this.getClass().getClassLoader().getResource("large_polygon.txt").getPath());
```

Comment:
`large_polygon.txt` looks complex. Do you think it would be worth benchmarking with smaller/simpler polygons?